### PR TITLE
Add `enable_progress_bar` to Trainer constructor

### LINKED ISSUE #9500: Deprecate `progress_bar_refresh_rate` from Trainer constructor
## Proposed refactoring or deprecation

Deprecate `progress_bar_refresh_rate` from Trainer constructor and use default refresh rate of 1 if no progress bar callback is passed

### Motivation

We are trying to minimize the number of arguments on Trainer constructor: #9006
`progress_bar_refresh_rate` makes more sense to be part of the Progress Bar callback (in fact it already is). If someone wants to customize the progress bar refresh rate, they must create a Progress Bar callback, just like if they wanted to customize any other part of the the progress bar.

### Pitch
Deprecate `progress_bar_refresh_rate` from Trainer constructor:
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/trainer.py#L117

Don't send `progress_bar_refresh_rate` here: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/trainer.py#L415
and update the relevant functions in callback_connector.py

### Considerations
One implication of this is there will no longer be a way to specify no progress bar callback - one will always be created by default (currently if you dont want a progress bar you can set `progress_bar_refresh_rate` equal to 0, which is a bit hacky IMO). Do people see this as a problem?

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?
Follow up from #9616 
Fixes #9500

Creates a new flag `enable_progress_bar` which allows users to disable the progress bar, since the previous method, setting `progress_bar_refresh_rate` to 0, was deprecated in #9616 
Also, replace `progress_bar_refresh_rate` with `enable_progress_bar` in the tests.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 1]
```diff
@@ -74,7 +75,20 @@ def on_trainer_init(
...
+            rank_zero_deprecation(
+                f"Setting `Trainer(progress_bar_refresh_rate={progress_bar_refresh_rate})` is deprecated in v1.5 and"
+                " will be removed in v1.7. Please pass `pytorch_lightning.callbacks.progress.ProgressBar` with"
+                " `refresh_rate` directly to the Trainer's `callbacks` argument instead."
+            )
+
+        if enable_progress_bar:
+            self.trainer._progress_bar_callback = self.configure_progress_bar(
+                progress_bar_refresh_rate, process_position
+            )
```
Comment (ananthsub) [inline pytorch_lightning/trainer/connectors/callback_connector.py:89]: what happens if progress_bar_refresh_rate=0 and enable_progress_bar=True? 
we still have to honor backwards compat for users setting progress_bar_refresh_rate=0 and expecting the progress bar to be disabled until v1.7

Comment (daniellepintz) [reply pytorch_lightning/trainer/connectors/callback_connector.py:89]: > what happens if progress_bar_refresh_rate=0 and enable_progress_bar=True?
same as before, we don't create a progress bar. the logic in configure_progress_bar isn't changing

**Review by awaelchli — APPROVED** [review]
we would need one more follow up to also update the occurrences in the docs

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 2]
```diff
@@ -82,7 +83,12 @@ def on_trainer_init(
                 " `refresh_rate` directly to the Trainer's `callbacks` argument instead."
             )
 
-        self.trainer._progress_bar_callback = self.configure_progress_bar(progress_bar_refresh_rate, process_position)
+        if enable_progress_bar:
```
Comment (carmocca) [inline pytorch_lightning/trainer/connectors/callback_connector.py:86]: Can you mention the newly added flag in the deprecation message above?

**Comment by daniellepintz** [comment]
@awaelchli added it to this PR!

File: CHANGELOG.md [hunk 3]
```diff
@@ -150,6 +150,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added `RichModelSummary` callback ([#9546](https://github.com/PyTorchLightning/pytorch-lightning/pull/9546))
 
 
+- Added `enable_progress_bar` to Trainer constructor ([#9664](https://github.com/PyTorchLightning/pytorch-lightning/pull/9664))
```
Comment (ananthsub) [inline CHANGELOG.md:153]: you could update the deprecation entry in the changelog for progress bar refresh rate to mention this new flag now

**Review by ananthsub — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
missing test for raising deprecation warning

**Comment by ananthsub** [comment]
> missing test for raising deprecation warning

@Borda a test was added here: https://github.com/PyTorchLightning/pytorch-lightning/blob/9148a13de0b121985bdf03e2832daf424f450ae5/tests/deprecated_api/test_remove_1-7.py#L230-L233 as part of #9616

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by awaelchli** [comment]
I'm asking myself now why did we prefix this with "enable_"? Just `progress_bar=True/False` would be shorter and the type makes it clear that a bool is expected.

**Comment by ananthsub** [comment]
> I'm asking myself now why did we prefix this with "enable_"? Just `progress_bar=True/False` would be shorter and the type makes it clear that a bool is expected.

Likely to avoid any confusion with ProgressBar as a callback instance. Similar to how checkpoint_callback is now typed as a bool, but sometimes people still pass a callback instance there

**Comment by daniellepintz** [comment]
@awaelchli Yeah it's exactly what @ananthsub said. See discussion here https://github.com/PyTorchLightning/pytorch-lightning/pull/9616#issuecomment-924387620
